nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/FetchHBaseRow.java

```diff
@@ -0,0 +1,407 @@
+
+        final RowSerializer rowSerializer = base64EncodeValues ? base64RowSerializer : regularRowSerializer;
+
+        final FetchHBaseRowHandler handler = destination.equals(DESTINATION_CONTENT.getValue())
+                ? new FlowFileContentHandler(flowFile, session, rowSerializer) : new FlowFileAttributeHandler(flowFile, session, charset, base64EncodeValues);
+
+        try {
+            final byte[] rowIdBytes = rowId.getBytes(StandardCharsets.UTF_8);
+            hBaseClientService.scan(tableName, rowIdBytes, rowIdBytes, columns, handler);
+        } catch (IOException e) {
+            getLogger().error("Unable to fetch row {} from  {} due to {}", new Object[] {rowId, tableName, e});
+            session.transfer(handler.getFlowFile(), REL_FAILURE);
+            return;
+        }
+
+        FlowFile handlerFlowFile = handler.getFlowFile();
+        if (!handler.handledRow()) {
+            getLogger().error("Row {} not found in {}, transferring to not found", new Object[] {rowId, tableName});
+            session.transfer(handlerFlowFile, REL_NOT_FOUND);
+            return;
+        }
+
+        if (getLogger().isDebugEnabled()) {
+            getLogger().debug("Fetched {} from {} with row id {}", new Object[]{handlerFlowFile, tableName, rowId});
+        }
+
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put(HBASE_TABLE_ATTR, tableName);
+        attributes.put(HBASE_ROW_ATTR, rowId);
+        if (destination.equals(DESTINATION_CONTENT)) {
+            attributes.put(CoreAttributes.MIME_TYPE.key(), "application/json");
+        }
+
+        handlerFlowFile = session.putAllAttributes(handlerFlowFile, attributes);
+
+        final String transitUri = "hbase://" + tableName + "/" + rowId;
+        if (destination.equals(DESTINATION_CONTENT)) {
+            session.getProvenanceReporter().fetch(handlerFlowFile, transitUri);
+        } else {
+            session.getProvenanceReporter().modifyAttributes(handlerFlowFile, "Added attributes to FlowFile from " + transitUri);
+        }
+
+        session.transfer(handlerFlowFile, REL_SUCCESS);
+    }
+
+    /**
+     * @param columnsValue a String in the form colFam:colQual,colFam:colQual
+     * @return a list of Columns based on parsing the given String
+     */
+    private List<Column> getColumns(final String columnsValue) {
+        final String[] columns = (columnsValue == null || columnsValue.isEmpty() ? new String[0] : columnsValue.split(","));
+
+        List<Column> columnsList = new ArrayList<>(columns.length);
+
+        for (final String column : columns) {
+            if (column.contains(":"))  {
+                final String[] parts = column.split(":");
+                final byte[] cf = parts[0].getBytes(StandardCharsets.UTF_8);
+                final byte[] cq = parts[1].getBytes(StandardCharsets.UTF_8);
+                columnsList.add(new Column(cf, cq));
```

Comment:
Nit: I'd move the `columnsList.add` call outside of the if/else, but that's just me.